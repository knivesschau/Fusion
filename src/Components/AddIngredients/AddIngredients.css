/* mobile styles first, min 320px */
#add-ingredient {
    font-size: .90rem;
    margin-top: 1.5em;
}

#add-ingredient {
    background-color: #A9C09B;
    text-transform: uppercase;
    color: black;
    font-weight: bold;
    transition: all .15s;
}

#remove-ingredient {
    background-color: #fac59d;
    text-transform: uppercase;
    color: black;
    font-weight: bold;
    transition: all .15s;
}

#add-ingredient:hover, #remove-ingredient:hover {
    cursor: pointer; 
    text-shadow: 0 0 32px rgba(255, 255, 255, 1);
    color: white;
    border-color: white;
}

#add-ingredient:active, #remove-ingredient:active {
    background-color: #881600;
}

#remove-ingredient {
    margin-top: .25em;
    margin-bottom: .25em;
}

#fuse-ingredients-add {
    font-size: .90rem;
    transition: border .3s ease-out; 
    width: 90%;
    margin: .45em;
}

#fuse-ingredients-add:focus {
    border: 10px solid #881600;
    border-style: none none none solid;
    background-color: white;
}

/* tablet and desktop styles */ 
@media only screen and (min-width: 768px) {
    #add-ingredient {
        font-size: 1.25rem;
        padding-top: .35em;
        padding-bottom: .35em;
        margin-top: .5em;
    }

    #remove-ingredient {
        font-size: 1rem;
        padding-top: .25em;
        padding-bottom: .25em;
    }

    #fuse-ingredients-add {
        width: 95%;
        margin: .5em;
        padding: .25em;
        font-size: 1.10rem;
    }
}

@media only screen and (min-width: 1200px) {
    #add-ingredient {
        font-size: 1.45rem;
    }

    #remove-ingredient {
        font-size: 1.15rem;
    }

    #fuse-ingredients-add {
        font-size: 1.25rem;
        padding: .5em;
    }
}