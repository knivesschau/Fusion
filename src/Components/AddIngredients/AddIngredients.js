import React, { useState } from 'react';
import './AddIngredients.css';

export default function AddIngredients() {
    const [ingredients, setIngredients] = useState({ ingredient: [] });

    function createInputs() {
        return ingredients.ingredient.map((ingredient, i) => 
            <li id="additional-ingredients" key={i}>
                <input
                    required
                    type="text"
                    id="fuse_ingredients"
                    name="fuse_ingredients"
                    value={ingredient || ""}
                    onChange={handleUpdate.bind(i)}/>

                <button type="button" id="remove-ingredient" onClick={removeInput.bind({i})}>Remove</button>
            </li>
        );
    };
    
    function handleUpdate(e) {
        let newIngredients = [...ingredients.ingredient];
        newIngredients[this] = e.target.value;

        setIngredients({
            ingredient: newIngredients
        });
    };

    const addInput = () => {
        setIngredients({
            ingredient: [...ingredients.ingredient, ""]
        });
    };

    const removeInput = () => {
        let newIngredients = [...ingredients.ingredient];
        newIngredients.splice(this, 1);

        setIngredients({
            ingredient: newIngredients
        });
    };

    return (
        <>
        {createInputs()}
        <button type="button" id="add-ingredient" onClick={addInput}>Add Ingredient</button>
        </>
    );
};
