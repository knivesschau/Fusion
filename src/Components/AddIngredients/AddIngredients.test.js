import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import AddIngredients from './AddIngredients';

describe ('AddIngredients Component', () => {
    it ('Renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<AddIngredients/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it ('Renders the complete AddIngredients Component', () => {
        const wrapper = shallow(<AddIngredients/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});