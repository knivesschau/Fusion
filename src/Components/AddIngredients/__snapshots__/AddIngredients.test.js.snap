// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddIngredients Component Renders the complete AddIngredients Component 1`] = `
<Fragment>
  <button
    id="add-ingredient"
    onClick={[Function]}
    type="button"
  >
    Add Ingredient
  </button>
</Fragment>
`;
